### PR TITLE
HPCC-13971 Improve timeouts and polling for ecl command

### PR DESCRIPTION
**Pull request by afishbeck**
Add more flexibility to control socket level timeouts.
--wait will now affect socket timeouts if not set explicitly.
--wait-connect can now be used to explicity control connect timeout.
--wait-read can now be used to explicity control read timeout.

Make polling less susceptible to socket and SOAP errors.

Improve verbose output during ecl-run.

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by richardkchapman** [comment]
@wangkx Please review

**Review by wangkx — COMMENTED** [review]
@afishbeck minor comment

File: ecl/eclcmd/eclcmd_core.cpp [hunk 1]
```diff
@@ -608,26 +612,74 @@ class EclCmdRun : public EclCmdWithEclTarget
...
+            case WUStateFailed:
+            case WUStateAborted:
+                return state;
+            }
+        }
+        catch (IJSOCK_Exception *E)
+        {
+            outputExceptionEx(*E);
+            E->Release();
+            fputs("Still polling...\n", stderr);
```
[the remark this answers is not in the document]
Comment (afishbeck) [reply ecl/eclcmd/eclcmd_core.cpp]: There is a case where no exception was thrown but state is not in:
case WUStateCompleted:
case WUStateFailed:
case WUStateAborted:
which will fall through and I don't want to say "still polling...".  It is still polling, but unless there are errors we don't need to display the extra info.

**Review by wangkx — APPROVED** [review]
@afishbeck looks fine.

File: ecl/eclcmd/eclcmd_core.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (wangkx) [reply ecl/eclcmd/eclcmd_core.cpp]: Does the WUStateUnknown mean "Still polling"?

[the remark this answers is not in the document]
Comment (afishbeck) [reply ecl/eclcmd/eclcmd_core.cpp]: Yes, but I don't want the words to appear in normal case.  I only say "still polling" if I've reported an error... so they know the error didn't cause polling to stop.

**Comment by afishbeck** [comment]
@richardkchapman this pull request can be merged

**Pull request merged**